assets/src/components/nav/navLinkData.ts

```diff
@@ -0,0 +1,38 @@
+import React from "react"
+import { mapModeForUser } from "../../util/mapMode"
+import { fullStoryEvent } from "../../helpers/fullStory"
+import { LadderIcon, MapIcon } from "../../helpers/icon"
+
+type HTMLElementProps = React.PropsWithoutRef<React.HTMLAttributes<HTMLElement>>
+
+export interface LinkData {
+  title: string
+  path: string
+  navIcon: React.JSXElementConstructor<HTMLElementProps>
+  onClick?: () => void
+}
+
+export const useNavLinkData: () => LinkData[] = () => {
```

Comment:
**question:** does this have to be a hook?

Reply:
This might be my lack of react fundamentals speaking, but the nav link data is dependent on what test groups you're in, so I don't _think_ it can just be a constant. 

Is there a different pattern that would make more sense in this context?

Reply:
I'm not sure if it's the right decision, but you _could_ implement this as a component I think.

Reply:
You can also make it a plain, non-hook function. Hooks have specific logic in React around triggering re-renders of components that I don't think is really relevant here.

Reply:
What makes this a hook as opposed to a plain old function?

I just tried writing `getNavLinkData()` as a plain old function, and the only difference was that it was named `get...` instead of `use....`. Is that all it takes?

Reply:
Yes, the `use` prefix is special: https://react.dev/learn/reusing-logic-with-custom-hooks#hook-names-always-start-with-use I'm not exactly sure what under-the-hood JavaScript magic React uses to accomplish this.

Reply:
For what it's worth, I'm _fairly_ sure the `use` prefix is just convention, and I don't think React does any compilation related to **hooks**, rather custom hooks often call react hooks to get reactivity (`useEffect`, `useState`, etc).

But because this function doesn't use any react hooks, it's probably good to follow convention and not use the `use` prefix.
> This convention guarantees that you can always look at a component and know where its state, Effects, and other React features might “hide”. For example, if you see a getColor() function call inside your component, you can be sure that it can’t possibly contain React state inside because its name doesn’t start with use. However, a function call like useOnlineStatus() will most likely contain calls to other Hooks inside!

Reply:
I think it does a little bit of stuff to provide runtime warnings when you break the Rules of Hooks: https://react.dev/warnings/invalid-hook-call-warning

Reply:
Also the ESLint plugin for React hooks will similarly complain about such things in CI. But yes, I think in practice it's mostly convention. In any event since this function doesn't need to do any "hook-y" things I'd advocate for renaming it.

Reply:
WELP!

And here I was thinking that I was being clever, but I guess I wasn't.

Just renamed it to `getNavLinkData`